streamlit_reports/streamlit_solver_hyundai.py

```diff
@@ -39,66 +39,461 @@
 placeholder_sales_plan_date = st.empty()
 placeholder_proposal_date = st.empty()
 placeholder_margins_date = st.empty()
 
-session_state = SessionState.get(first_run_flag=0, run_id=0, run_id_scores=0, save_button_pressed_flag=0, model='', brand='',
-                                 daysinstock_score_weight=score_weights['Avg_DaysInStock_Global_normalized'],
-                                 sel_margin_score_weight=score_weights['TotalGrossMarginPerc_normalized'],
-                                 sel_margin_ratio_score_weight=score_weights['MarginRatio_normalized'],
-                                 sel_qty_sold_score_weight=score_weights['Sum_Qty_CHS_normalized'],
-                                 sel_proposals_score_weight=score_weights['Proposals_VDC_normalized'],
-                                 sel_oc_stock_diff_score_weight=score_weights['Stock_OC_Diff_normalized'],
-                                 sel_co2_nedc_score_weight=score_weights['NEDC_normalized'],
-                                 sel_configurator_count_score_weight=score_weights['Configurator_Count_normalized'])
+session_state = SessionState.get(
+    first_run_flag=0, 
+    run_id=0, 
+    run_id_scores=0, 
+    save_button_pressed_flag=0, 
+    model='',
+    brand='',
+    daysinstock_score_weight=score_weights['Avg_DaysInStock_Global_normalized'],
+    sel_margin_score_weight=score_weights['TotalGrossMarginPerc_normalized'],
+    sel_margin_ratio_score_weight=score_weights['MarginRatio_normalized'],
+    sel_qty_sold_score_weight=score_weights['Sum_Qty_CHS_normalized'],
+    sel_proposals_score_weight=score_weights['Proposals_VDC_normalized'],
+    sel_oc_stock_diff_score_weight=score_weights['Stock_OC_Diff_normalized'],
+    sel_co2_nedc_score_weight=score_weights['NEDC_normalized'],
+    sel_configurator_count_score_weight=score_weights['Configurator_Count_normalized'],
+    #
+    sales_plan_changes = {},
+    change_co2_target = False,
+    num_added_electrics = 0
+    )
 
 temp_cols = ['Avg_DaysInStock_Global', 'Avg_DaysInStock_Global_normalized', '#Veículos Vendidos', 'Sum_Qty_CHS_normalized', 'Proposals_VDC', 'Proposals_VDC_normalized', 'Margin_HP', 'TotalGrossMarginPerc', 'TotalGrossMarginPerc_normalized', 'MarginRatio', 'MarginRatio_normalized', 'OC', 'Stock_VDC', 'Stock_OC_Diff', 'Stock_OC_Diff_normalized', 'NEDC', 'NEDC_normalized', 'Config_Total', 'Config_Total_normalized']
 total_months_list = ['Jan', 'Fev', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
 
+# choose between Order Optimization or CO2 simulator
+main_function = st.selectbox(
+    label = 'Aplicação:', 
+    options = ['Simulador CO2', 'Order Optimization'],
+    help = 'Por favor escolha a aplicação a usar'
+    )
+
+def co2_simulator():
+    from sympy import solve, Poly, Eq, Function, exp
+    from sympy.abc import x, y, z, a, b
+    from datetime import datetime
+    
+    st.markdown("<h1 style='text-align: center;'>Simulador de CO2</h1>", unsafe_allow_html=True)
```

Comment:
Adiciona também a marca. Se for Hyundai, mete Hyundai, se forem as duas, mete Importador.

Reply:
feito